docs/develop/maintenance/node.md

```diff
@@ -1,3 +1,375 @@
-# Node / TypeScript Dependency Maintenance
+# Node.js / TypeScript Dependency Maintenance
 
-(Placeholder – to be populated later.)
+This guide describes the process for updating and validating Node.js and TypeScript dependencies in the monorepo. It focuses on using pnpm workspaces with a centralized `package.json` and ensuring consistent, reproducible builds across Angular, React, and Node.js applications.
+
+## Overview
+
+Node.js dependencies are managed primarily via the root `package.json` file in conjunction with pnpm workspaces. Individual projects reference shared dependencies from the root, ensuring:
+
+- Version alignment across all applications and libraries
+- Simplified upgrades and security remediation
+- Reduced dependency drift and bundle duplication
+- Efficient disk usage through symlinked node_modules
+
+### Key Files
+
+| File / Path                                  | Role / Purpose                                                    | Maintenance Notes                                                 |
+| -------------------------------------------- | ----------------------------------------------------------------- | ----------------------------------------------------------------- |
+| `package.json` (root)                        | Central dependency management for all TypeScript/Node.js projects | Primary source of truth for versions                              |
+| `pnpm-workspace.yaml`                        | Defines workspace packages and exclusions                         | Keep aligned with project structure                               |
+| `tsconfig.base.json`                         | TypeScript path mapping and compiler configuration                | Update path mappings when adding/removing libraries               |
+| `nx.json`                                    | Nx workspace configuration and task execution                     | Configure build, test, and lint tasks                             |
+| `apps/*/project.json`, `libs/*/project.json` | Individual project configurations                                 | Should reference root dependencies via Nx or workspace resolution |
+| `pnpm-lock.yaml`                             | Lock file ensuring reproducible installations                     | Automatically updated; commit changes                             |
+| `jest.config.ts`, `jest.preset.js`           | Testing framework configuration                                   | Align with testing library versions                               |
+| `stylelint.config.mjs`, `tsconfig.base.json` | Linting and TypeScript configuration                              | Keep linter versions synchronized                                 |
+
+!!! note "Workspace dependency resolution"
+
+    Individual projects within the workspace automatically inherit dependencies from the root `package.json`. Generally, project-specific `package.json` files should not be used.
```

Comment:
See previous comment. A publishable library would typically require a `package.json`, though the library may still tap into the shared `node_modules`. I believe there is a risk of package definition drift if the local `package.json` is not regularly reviewed and updated. It would be great if the local `package.json` could be generated based on the root `package.json` and the libraries effectively used by the publishable library. We should research once what the recommended approach is to maintaining local `package.json`.